frontend/src/views/admin/assignments/confirm-with-dialog-action-button.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { ActionButton } from "../../../components/action-buttons";
+import { OfferConfirmationDialog } from "./offer-confirmation-dialog";
+import { Assignment } from "../../../api/defs/types";
+
+export function ConfirmWithDialogActionButton(props: {
+    buttonIcon: Object;
+    buttonEnable: boolean;
+    actionConfirmation: () => void;
+    buttonText: string;
+    selectedAssignments: Assignment[];
+    dialogVisible: boolean;
+    dialogSetVisible: Function;
+    actionCallback: Function;
+    dialogTitle: string;
+    dialogBody: string;
+    dialogConfirmation: string;
```

Comment:
There are way too many props here. See how few you can reduce it to.

`dialogConfirmation` <- bad name